apps/frontend/src/components/searchBar.tsx

```diff
@@ -0,0 +1,252 @@
+import { useState, ChangeEvent, useEffect, useCallback } from "react";
+import Filters from "./filters";
+import { useNavigate } from "react-router-dom";
+import * as React from "react";
+
+interface SearchObject {
+  query: string;
+  minPrice: number;
+  maxPrice: number;
+  status: string;
+  searchType: string;
+  latitude: number;
+  longitude: number;
+  sort: string;
+  page: number;
+  limit: number;
+}
+
+interface SearchBarProps {
+  onSearch: () => void;
+  sortBy: string;
+}
+
+const SearchBar: React.FC<SearchBarProps> = ({ onSearch, sortBy }) => {
+  const classes = useStyles();
+  const navigate = useNavigate();
+
+  const [showFilters, setShowFilters] = useState(false);
+  const [searchInput, setSearchInput] = useState("");
+  const [filters, setFilters] = useState<SearchObject>({
+    query: "",
+    minPrice: 0,
+    maxPrice: 100000,
+    status: "AVAILABLE",
+    searchType: "LISTING",
+    latitude: 0,
+    longitude: 0,
+    sort: "RELEVANCE",
+    page: 1,
+    limit: 20,
+  });
+
+  const toggleFilters = () => {
+    setShowFilters(!showFilters);
+  };
+
+  const handleSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(event.target.value);
+  };
+
+  const handleFilterChange = (newFilters: Partial<SearchObject>) => {
+    setFilters((prevFilters) => ({ ...prevFilters, ...newFilters }));
+  };
+
+  const handleSearch = useCallback(() => {
+    const searchObject: SearchObject = {
+      ...filters,
+      query: searchInput,
+      sort: sortBy,
+    };
+    console.log("Search Object:", searchObject);
+    onSearch();
+    // API CALL HERE???
+  }, [filters, searchInput, sortBy, onSearch]);
+
+  useEffect(() => {
+    if (sortBy !== "") {
+      handleSearch();
+    }
+  }, [sortBy, handleSearch]);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter") {
+      handleSearch();
+    }
+  };
+
+  const handleMessageRoute = () => {
+    navigate("/messages");
+  };
+
+  const handleAccountRoute = () => {
+    navigate("/user");
+  };
+
+  const handleReload = () => {
+    navigate("/");
+    window.location.reload();
+  };
+
+  const handleListingRoute = () => {
+    navigate("/listing/view");
+  };
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const isDesktop = useMediaQuery("(min-width:1240px)");
+  const isSmallPage = useMediaQuery("(min-width:740px)");
+
+  return (
+    <Grid
+      container
+      direction="row"
+      alignItems="center"
+      justifyContent="center"
+      spacing={2}
+      sx={{
+        backgroundColor: "#f5f5f5",
+        boxShadow: "0px 4px 6px #808080",
+        paddingBottom: "10px",
+      }}
+    >
+      <Grid item>
+        {isSmallPage ? (
```

Comment:
`{isSmallPage &&
   <Box (or Grid maybe, not sure how MUI does it)>
     <Button> ... </Button>
   </Box>
 }
`

Reply:
Updated I kept the grid outside the if statement to keep the spacing consistent